app/controllers/user_training_controller.rb

```diff
@@ -0,0 +1,30 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+class UserTrainingController < ApplicationController
+  def index
+    config = Talentlms::Config.first
+    if current_user.training_completed? || !current_user.training_required? || config.nil?
```

Comment:
I think the order of these should be reversed.  Looking to see if there's a config is very cheap.  `training_required?` is also just a method on the users.  Presumably `training_completed?` needs to hit the API.